tests/queries/0_stateless/01946_test_zstd_decompression_with_escape_sequence_at_the_end_of_buffer.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+CUR_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CUR_DIR"/../shell_config.sh
+
+
+# See 01658_read_file_to_string_column.sh
+user_files_path=$(clickhouse-client --query "select _path,_file from file('nonexist.txt', 'CSV', 'val1 char')" 2>&1 | grep Exception | awk '{gsub("/nonexist.txt","",$9); print $9}')
+mkdir -p ${user_files_path}/
+cp $CUR_DIR/data_zstd/test_01946.zstd ${user_files_path}/
+
+${CLICKHOUSE_CLIENT} --multiline --multiquery --query "
+set max_read_buffer_size = 65536;
```

Comment:
It reproduced in hdfs because it coincided that reading from hdfs has buf size 65536 and with current data file - this size is perfect to have escape sequence right at the end of read buffer. Parsing escape sequence in the middle was stopped because nextImpl resulted in empty buffer though it was not eof yet. This resulted in crash in case use_parallel_parsing = 1 and in failed parsing in case use_parallel_parsing = 0.